asana/asana.go

```diff
@@ -401,8 +413,22 @@ func (c *Client) request(ctx context.Context, method string, path string, data i
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode == http.StatusUnauthorized {
+	// See https://asana.com/developers/documentation/getting-started/errors
+	switch resp.StatusCode {
+	case http.StatusBadRequest:
+		return nil, ErrBadRequest
+	case http.StatusUnauthorized:
 		return nil, ErrUnauthorized
+	case http.StatusPaymentRequired:
+		return nil, ErrPaymentRequired
+	case http.StatusForbidden:
+		return nil, ErrForbidden
+	case http.StatusNotFound:
+		return nil, ErrNotFound
+	case http.StatusTooManyRequests:
+		return nil, ErrThottled
+	case http.StatusInternalServerError:
+		return nil, ErrInternal
```

Comment:
Asana doesn't send JSON for errors. So this uses response codes and returns a well-formed error instead of trying to parse the response. 

If the JSON fails to parse, it'll still return an error below, which is ok, but for a different class of error.